common.go

```diff
@@ -7,9 +7,11 @@ import (
 	"github.com/containers/image/types"
 )
 
-func getCopyOptions(reportWriter io.Writer) *cp.Options {
+func getCopyOptions(reportWriter io.Writer, credsContext *types.SystemContext) *cp.Options {
 	return &cp.Options{
-		ReportWriter: reportWriter,
+		ReportWriter:   reportWriter,
+		SourceCtx:      credsContext,
+		DestinationCtx: credsContext,
```

Comment:
we shouldn't be setting system context, which holds credentials, to both Source and Destination. It's a security issue if we send credentials where we're not supposed to (e.g. the credentials are just for Source or just for Destination, it's a credentials leak otherwise...).